cpp/src/reshape/byte_cast.cu

```diff
@@ -101,11 +101,21 @@ std::unique_ptr<cudf::column> byte_list_conversion::operator()<string_view>(
   auto strings_count = input_strings.size();
   if (strings_count == 0) return cudf::empty_like(input_column);
 
-  auto contents = std::make_unique<column>(input_column, stream, mr)->release();
+  auto col_content = std::make_unique<column>(input_column, stream, mr)->release();
+  auto contents =
+    col_content.children[strings_column_view::chars_column_index].release()->release();
+  auto data      = contents.data.release();
+  auto null_mask = contents.null_mask.release();
+  auto uint8_col = std::make_unique<column>(data_type{type_id::UINT8},
+                                            data->size(),
+                                            std::move(*data),
+                                            std::move(*null_mask),
+                                            UNKNOWN_NULL_COUNT);
```

Comment:
Should we fetch the null count before releasing the data, and pass it here? We should try to retain any information that may have been previously computed.

Generally, I think we're going to move away from lazily computing null counts as we work on adding streams to the public API. There are some hiccups with the lazy approach and streams, but primarily we aren't being as efficient as possible wrt avoiding calls to the `null_count` kernel because we aren't tracking information that we already may have computed once earlier.

Reply:
I like the idea of moving away form lazily computed null counts and am happy to change here if we want to start now. In this case, it made sense for me to defer the null counting because byte cast is often use as an intermediate type for reading, writing, and hashing. Those are usually indifferent to null count and it made sense to keep it lazily computed while we still supported it.